### ISSUE TITLE
Add Dockerfile

### ISSUE THREAD

**Issue by leonbeon** [cooklang/cookcli#90]
I think adding a Dockerfile would really benefit this project! Here is the one I currently use for a Raspberry Pi, based on Alpine Linux.

```Dockerfile
from alpine:3.18

ARG DOWNLOAD_URL="https://github.com/cooklang/cookcli/releases/latest/download/cook-aarch64-unknown-linux-musl.tar.gz"

# Set workdir to /app
WORKDIR /app

# Download CookCLI file
ADD ${DOWNLOAD_URL} .

# Untar binary
RUN tar -xvf cook-*

# Remove tar
RUN rm *.tar.gz

# Expose default port (9080)
EXPOSE 9080

# Run Server
ENTRYPOINT ["./cook", "server", "--host", "./recipes"]
```

You could overwrite the download URL with `--build-arg`, currently it points to the latest build of the aarch64 version. Also, I'm not sure if logs would be redirected to `docker logs` output, since it seems like the `cook` binary doesn't output anything anyway?

Also, I spent hours realising that the `--host` argument is missing. [Maybe add that to the docs](https://cooklang.org/cli/help/#server)?

**Comment by dubadub** [comment]
@leonbeon I'm not certain I understand the motivation for this docker file. I guess the goal is to have a server running, but why not to use default system tools for that (systemd for Linux, launchd for MacOS, probably something exists for Windows too)?

**Comment by dubadub** [comment]
Good point about the docs, it still refers old Swift base CLI, need to update.

**Comment by leonbeon** [comment]
I think the benefit of having a Dockerfile on the top level of the repo is that [you can build directly from GitHub](https://stackoverflow.com/a/34120804). But you're right, using Docker is just one of many ways to build this. Then again, why not also add a systemd config for easy deployment?

**Comment by kewiha** [comment]
The use case for having a Dockerfile like the one above in the repo makes perfect sense to me. I'm setting up a publicly accessible cookcli server and already have a podman server this could run on. I think self-hosting/homelab people frequently have a docker-compatible server available that they know how to use.

There are many ways to do the same thing, but having a dockerfile in this repo seems worth the effort. I'd be happy to submit a PR if you're open to merging it, @dubadub.

**Comment by dubadub** [comment]
Certainly, let's add
